operator/controllers/instance_controller.go

```diff
@@ -184,6 +192,36 @@ func (r *InstanceReconciler) valuesSecret(
 	return false, nil
 }
 
+// mergeValues merges the user-provided values and the default and static values provided by the
+// plan. The static values take precedence over the user-provided values, which in turn takes
+// precedence over the default values.
+func mergeValues(
```

Comment:
I _think_ [mergo](https://pkg.go.dev/github.com/imdario/mergo) is supposed to do similar things; not sure if that's worth pulling in here though.

Reply:
Yeah, I considered mergo and decided to just copy this single function from Helm instead. It does what it's supposed to do.